src/metabase/db/liquibase.clj

```diff
@@ -11,19 +13,37 @@
            [liquibase.database Database DatabaseFactory]
            liquibase.database.jvm.JdbcConnection
            liquibase.exception.LockException
-           liquibase.resource.ClassLoaderResourceAccessor
-           liquibase.sqlgenerator.SqlGeneratorFactory
-           metabase.db.liquibase.MetabaseMySqlCreateTableSqlGenerator))
+           liquibase.resource.ClassLoaderResourceAccessor))
 
-(.register (SqlGeneratorFactory/getInstance) (MetabaseMySqlCreateTableSqlGenerator.))
+;; register our custom MySQL SQL generators
+(liquibase.mysql/register-mysql-generators!)
+
+;; Liquibase uses java.util.logging (JUL) for logging, so we need to install the JUL -> Log4j2 bridge which replaces the
+;; default JUL handler with one that "writes" log messages to Log4j2. (Not sure this is the best place in the world to
+;; do this, but Liquibase is the only thing using JUL directly.)
+;;
+;; See https://logging.apache.org/log4j/2.x/log4j-jul/index.html for more information.
+(org.apache.logging.log4j.jul.Log4jBridgeHandler/install true nil true)
+
+;; Liquibase logs a message for every ChangeSet directly to standard out -- see
+;; https://github.com/liquibase/liquibase/issues/2396 -- but we can disable this by setting the ConsoleUIService's
+;; output stream to the null output stream
+(doto ^liquibase.ui.ConsoleUIService (.getUI (liquibase.Scope/getCurrentScope))
+  ;; we can't use `java.io.OutputStream/nullOutputStream` here because it's not available on Java 8
+  (.setOutputStream (java.io.PrintStream. (org.apache.commons.io.output.NullOutputStream.))))
```

Comment:
should these be imported at the top of the file? not totally sure what our conventions are for java classes

Reply:
The only thing `import` does is let you use the class without fully qualifying it -- referencing the class will load it either way. Since we're only using all of these classes once I didn't think it was really worth it